src/controllers/FlowController.js

```diff
@@ -1,406 +1,389 @@
-import FlowStage from '../models/FlowStage.js';
-import User from '../models/User.js';
-import FlowUser from '../models/FlowUser.js';
-import { QueryTypes } from 'sequelize';
-import Database from '../database/index.js';
-
-export async function getMailContents() {
+import Flow from "../models/Flow.js";
+import Stage from "../models/Stage.js";
+import FlowStage from "../models/FlowStage.js";
+import User from "../models/User.js";
+import FlowUser from "../models/FlowUser.js";
+import { QueryTypes } from "sequelize";
+import Database from "../database/index.js";
+
+class FlowController {
+  async index(req, res) {
+    const flows = await Flow.findAll();
+
+    if (!flows) {
+      return res.status(401).json({ error: "Não Existem fluxos" });
+    } else {
+      return res.json({ Flows: flows });
+    }
+  }
+
+  // async getMailContentsEndpoint(req, res) {
+  //     const contents = await getMailContents();
+  //     if (contents.error) {
+  //         return res.status(500).json(contents);
+  //     } else {
+  //         return res.status(200).json(contents);
+  //     }
+  // }
+
+  async indexForFrontend(req, res) {
     try {
-        const mailContents = await Database.connection.query(
-            "select \
-            f.\"idFlow\" as id_flow, \
-            f.\"name\" as flow, \
-            p.record as process_record, \
-            p.nickname as process, \
-            s.\"idStage\" as id_stage, \
-            s.\"name\" as stage, \
-            p.\"effectiveDate\" as start_date, \
-            s.duration as stage_duration, \
-            u.email as email, \
-            extract(day from (current_timestamp - p.\"effectiveDate\")) - cast(s.duration as integer) as delay_days \
-            from \
-                users u \
-            join \"flowUser\" fu on \
-                fu.cpf = u.cpf \
-            join \"flowProcess\" fp on \
-                fp.\"idFlow\" = fu.\"idFlow\" \
-            join process p on \
-                p.record = fp.record \
-            join stage s on \
-                s.\"idStage\" = p.\"idStage\" \
-            join flow f on \
-                f.\"idFlow\" = fp.\"idFlow\" \
-            where \
-                extract(day from (current_timestamp - p.\"effectiveDate\")) > cast(s.duration as integer)",
-            {
-                type: QueryTypes.SELECT
+      const flows = await Flow.findAll();
+      let flowsWithSequences = [];
+      for (const flow of flows) {
+        const flowStages = await FlowStage.findAll({
+          where: {
+            idFlow: flow.idFlow,
+          },
+        });
+
+        let sequences = [];
+        let stages = [];
+
+        if (flowStages.length > 0) {
+          for (let i = 0; i < flowStages.length; i++) {
+            sequences.push({
+              from: flowStages[i].idStageA,
+              commentary: flowStages[i].commentary,
+              to: flowStages[i].idStageB,
+            });
+            if (!stages.includes(flowStages[i].idStageA)) {
```

Comment:
Uma aleterativa a esse for seria [const of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)

Reply:
Eu e o Pedro não fizemos essa parte.

Reply:
Então podem deixar como está. Não é complicado fazer a mudança, mas tem que mudar várias linhas.
Já tem uma versão convertida de um código muito parecido na branch `refactor/preparacao_banco_de_dados`